src/libs/Tokenizer.js

```diff
@@ -0,0 +1,95 @@
+
+
+var theTokenizer;
+
+class Tokenizer {
+
+    constructor(program, literalsList) {
+        this.program = program;
+        this.literals = literalsList;
+        this.tokens = [];
+        this.currentToken = 0;
+        this.tokenize();
+    }
+
+    tokenize() {
+        let tokenizedProgram = this.program;
+        tokenizedProgram = tokenizedProgram.replace(/\n/g, "");
+
+        // extracts replacing them with _$_, stores them in an array
+        // to be added back in later.
+        var texts = [...tokenizedProgram.matchAll(/".*"/g)];
+        tokenizedProgram = tokenizedProgram.replace(/".*"/g, "_$_");
```

Comment:
Similar to comment above, this is replacing every string enclosed by quotation marks in our program with the string `_$_`, but we don't have anything in our grammar within quotation marks. Maybe I'm missing to see something...?

Reply:
Used for text, see the code examples you provided. I think this should actually be in quotes since its kinda intuitive and makes this easier.

Reply:
Totally right. Thanks for pointing out.